Add documentation for ICorProfilerInfo13 methods and enumeration. (#30131)
dotnet/docs: docs/framework/unmanaged-api/profiling/icorprofilerinfo13-createhandle-method.md

---
description: "Learn more about: ICorProfilerInfo13::CreateHandle Method"
title: "ICorProfilerInfo13::CreateHandle Method"
ms.date: "03/19/2021"
api_name: 
  - "ICorProfilerInfo13.CreateHandle"
api_location: 
  - "coreclr.dll"
  - "corprof.idl"
api_type: 
  - "COM"
---
# ICorProfilerInfo13::CreateHandle Method

> Create a handle wrapping the given object.

Reviewer: ```suggestion
Creates a handle that wraps a specified object.
```

  
## Syntax  
  
```cpp  
    HRESULT CreateHandle(
                [in] ObjectID            object,
                [in] COR_PRF_HANDLE_TYPE type,
                [out] ObjectHandleID*    pHandle);
```  
  
## Parameters

`object`
[in] The object reference to wrap with a handle.

`type`
[in] The type of handle to create.

`pHandle`

> [out] A caller provided pointer that will be filled with the handle created to wrap the given `object`.

Reviewer: ```suggestion
[out] A caller-provided pointer that will point to the handle created to wrap the specified `object`.
```

## Remarks

> Here are the expected usages depending on the given `COR_PRF_HANDLE_TYPE` value:

Reviewer: ```suggestion
Following are the expected usages depending on the specified `COR_PRF_HANDLE_TYPE` value:
```

>  - COR_PRF_HANDLE_TYPE_WEAK: monitors if an object stays in memory over time. If the wrapped object has been collected, [ICorProfilerInfo13::GetObjectIDFromHandle](icorprofilerinfo13-get-objectid-from-handle-method.md) will return a null `ObjectID` for its wrapping weak handle.

Reviewer: ```suggestion

- `COR_PRF_HANDLE_TYPE_WEAK`: Monitors if an object stays in memory over time. If the wrapped object has been collected, [ICorProfilerInfo13::GetObjectIDFromHandle](icorprofilerinfo13-get-objectid-from-handle-method.md) will return a null `ObjectID`.
```

Author: ✅ Done

>  - COR_PRF_HANDLE_TYPE_STRONG: enforces that an object survives garbage collections even though no other object references it.

Reviewer: ```suggestion
- `COR_PRF_HANDLE_TYPE_STRONG`: Enforces that an object survives garbage collections even though no other object references it.
```

> - `COR_PRF_HANDLE_TYPE_STRONG`: Enforces that an object survives garbage collections even though no other object references it.

Reviewer: Somehow it looks like my suggestions to fix these warnings were lost.

Author: > Somehow it looks like my suggestions to fix these warnings were lost.

@gewarren: Let me know if the last push fixed the indentation issue

>  - COR_PRF_HANDLE_TYPE_PINNED: same as a strong handle while ensuring that the object stays at the same address in memory during garbage collections.

Reviewer: ```suggestion
- `COR_PRF_HANDLE_TYPE_PINNED`: Same as a strong handle but also ensures that the object stays at the same address in memory during garbage collections.
```

>  To ensure that the `object` reference is valid, this method has to be called from a `ICorProfilerCallback` method such as [ICorProfilerCallback::ObjectAllocated](icorprofilercallback-object-allocated-method.md). It is not allowed to call `CreateHandle` from an EventPipe asynchronous listener. The object received via an event payload might have been disposed or moved in memory if a garbage collection occured between the time the event has been emitted and when it has been received.

Reviewer: ```suggestion
 To ensure that the `object` reference is valid, this method has to be called from a `ICorProfilerCallback` method, such as [ICorProfilerCallback::ObjectAllocated](icorprofilercallback-object-allocated-method.md). You cannot call `CreateHandle` from an EventPipe asynchronous listener. The object received via an event payload might have been disposed or moved in memory if a garbage collection occurred between the time the event was emitted and when it was received.
```

Author: ✅ Done

> Do not forget to call [ICorProfilerInfo13::DestroyHandle](icorprofilerinfo13-destroyhandle-method.md) once a created handle is no more needed.

Reviewer: ```suggestion
Do not forget to call [ICorProfilerInfo13::DestroyHandle](icorprofilerinfo13-destroyhandle-method.md) once a created handle is no longer needed.
```

---
description: "Learn more about: ICorProfilerInfo13::CreateHandle Method"
title: "ICorProfilerInfo13::CreateHandle Method"
ms.date: "03/19/2021"
api_name: 
  - "ICorProfilerInfo13.CreateHandle"
api_location: 
  - "coreclr.dll"
  - "corprof.idl"
api_type: 
  - "COM"
---
# ICorProfilerInfo13::CreateHandle Method

Creates a handle that wraps a specified object.
  
## Syntax  
  
```cpp  
    HRESULT CreateHandle(
                [in] ObjectID            object,
                [in] COR_PRF_HANDLE_TYPE type,
                [out] ObjectHandleID*    pHandle);
```  
  
## Parameters

`object`
[in] The object reference to wrap with a handle.

> `type`

Author: ✅ done

[in] The type of handle to create.

> `pHandle`

Author: ✅ done
